aten/src/ATen/native/cuda/UpSample.cuh

```diff
@@ -0,0 +1,254 @@
+#include <ATen/ATen.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+
+#include <math.h>
+
+namespace at {
+namespace native {
+
+/* TODO: move this to a common place */
+template <typename scalar_t>
+__device__ inline scalar_t min(scalar_t a, scalar_t b) {
+  return ((a) < (b)) ? (a) : (b);
```

Comment:
nit (you don't have to fix this, but FYI), you don't need the extra parentheses if you aren't writing  a macro.

Reply:
thanks @ezyang for the explanation!